tests/integration/high_availability/test_upgrade_rollback_incompat.py

```diff
@@ -124,9 +132,15 @@ async def test_upgrade_to_failling(
 async def test_rollback(ops_test, continuous_writes) -> None:
     application = ops_test.model.applications[MYSQL_APP_NAME]
 
-    sub_regex_older_snap = "s/CHARMED_MYSQL_SNAP_REVISION.*/CHARMED_MYSQL_SNAP_REVISION = 69/"
-    src_patch(sub_regex=sub_regex_older_snap, file_name="src/constants.py")
-    charm = await charm_local_build(ops_test, refresh=True)
+    snap_revisions = pathlib.Path("snap_revisions.json")
+    with snap_revisions.open("r") as file:
+        old_revisions: dict = json.load(file)
+    new_revisions = old_revisions.copy()
+    # TODO: mark as amd64 only or support arm64
```

Comment:
This is two fold:
- for now mark as amd64 only
- when there's an incompatible version upgrade from mysql, we re-add it with the previous snap. (e.g. amd64 snap rev69 is mysql-8.0.34, and mysql-8.0.35 introduced breaking changes to the data dir)